drivers/adc/ad7606/ad7606.c

```diff
@@ -1122,9 +1133,13 @@ int32_t ad7606_init(struct ad7606_dev **device,
 		return -ENOMEM;
 
 	dev->device_id = init_param->device_id;
-	dev->num_channels = ad7606_chip_info_tbl[dev->device_id].num_channels;
-	dev->max_dout_lines = ad7606_chip_info_tbl[dev->device_id].max_dout_lines;
-	if (ad7606_chip_info_tbl[dev->device_id].has_registers)
+	info = &ad7606_chip_info_tbl[dev->device_id];
+	printf("Initializing device %s, num-channels %u SDI lines %u\n",
+	       info->name, info->num_channels, 1 << info->max_dout_lines);
+
+	dev->num_channels = info->num_channels;
+	dev->max_dout_lines = info->max_dout_lines;
```

Comment:
maybe it is worth passing directly the chip info structure to the device descriptor (as it is a common approach for linux drivers too). but i guess this is subject for another patch.

Reply:
yeah, that would have been a good idea from the start;

i may add it later.